### ISSUE TITLE
[tune/placement groups] trial resources not being properly allocated in tune.run

### ISSUE THREAD

**Issue by ANarayan** [ray-project/ray#14759]
Ray Version: Linux, Python3.7 Nightly Version - commit: 34b5781ae0d46b769e8bac400e0cbad6c60c8fcd

I am running a ray tune experiment on in an interactive slurm session w/3GPUs. I have my specified my `resources_per_trial={'cpu' : 2, 'gpu' : 1}` and am running two trials concurrently. However, I am seeing an unexpected behavior where two trials are being allocated to the same GPU.

Here are the output of `ray.available_resources()` prior to the `tune.run` call:
`Available resources: {'accelerator_type:GTX': 1.0, 'memory': 72773644084.0, 'GPU': 3.0, 'node:172.24.67.146': 1.0, 'object_store_memory': 35474418892.0, 'CPU': 32.0}`

Inside two different trials, the output of `ray.get_gpu_ids()` and `ray.get_resource_ids()` is as follows:
```
(pid=1040823) GPU ids: ['0']
(pid=1040823) Resource ids: {'CPU_group_4d2371f34840192a944062821327f2ea': [(0, 2.0)], 'GPU_group_0_4d2371f34840192a944062821327f2ea': [(0, 1.0)], 'GPU_group_4d2371f34840192a944062821327f2ea': [(0, 1.0)], 'CPU_group_0_4d2371f34840192a944062821327f2ea': [(0, 2.0)]}
(pid=1040826) GPU ids: ['0']
(pid=1040826) Resource ids: {'CPU_group_4acb05de24ad14cfada9a074c6f6e5f5': [(0, 2.0)], 'GPU_group_4acb05de24ad14cfada9a074c6f6e5f5': [(0, 1.0)], 'GPU_group_0_4acb05de24ad14cfada9a074c6f6e5f5': [(0, 1.0)], 'CPU_group_0_4acb05de24ad14cfada9a074c6f6e5f5': [(0, 2.0)]}
```

I was able to avoid the issue by setting the following environment variable: `os.environ["TUNE_PLACEMENT_GROUP_AUTO_DISABLED"] = "1"`

cc: @richardliaw

**Comment by ericl** [comment]
This seems to be a regression as of ray==1.2.0

```
import ray
import time

ray.init(num_gpus=2)

@ray.remote(num_gpus=1, num_cpus=0)
def f():
    import os
    return os.environ["CUDA_VISIBLE_DEVICES"]

pg1 = ray.util.placement_group([{"GPU": 1}])
pg2 = ray.util.placement_group([{"GPU": 1}])
print(ray.get(f.options(placement_group=pg1).remote()))
print(ray.get(f.options(placement_group=pg2).remote()))
```

1.2:
0
0

1.1:
1
0

**Comment by clay4444** [comment]
> This seems to be a regression as of ray==1.2.0
> 
> ```
> import ray
> import time
> 
> ray.init(num_gpus=2)
> 
> @ray.remote(num_gpus=1, num_cpus=0)
> def f():
>     import os
>     return os.environ["CUDA_VISIBLE_DEVICES"]
> 
> pg1 = ray.util.placement_group([{"GPU": 1}])
> pg2 = ray.util.placement_group([{"GPU": 1}])
> print(ray.get(f.options(placement_group=pg1).remote()))
> print(ray.get(f.options(placement_group=pg2).remote()))
> ```
> 
> 1.2:
> 0
> 0
> 
> 1.1:
> 1
> 0

hi, @ericl, I do actually not get the meaning of this code statement, what's the stand for of the env `CUDA_VISIBLE_DEVICES `

**Comment by wuisawesome** [comment]
@clay4444  those are the ID's of the gpus (the first GPU has id 0, the second has id 1). If CUDA_VISIBLE_DEVICES="0" for both tasks, then they will try to use the same GPU.

**Comment by clay4444** [comment]
> @clay4444 those are the ID's of the gpus (the first GPU has id 0, the second has id 1). If CUDA_VISIBLE_DEVICES="0" for both tasks, then they will try to use the same GPU.

Yeah, I got it, thanks~

**Comment by amogkam** [comment]
@wuisawesome @clay4444 can this be prioritized asap please. This is blocking all Tune-related GPU testing.

**Comment by wuisawesome** [comment]
@richardliaw let's keep both the P1 and P0 labels (we always label core bugs with both to make it easier to filter/track)

**Comment by clay4444** [comment]
hi, @amogkam  I have fixed this issue in this pr:  #14891  , maybe you can task a look for this, thanks~

**Comment by wuisawesome** [comment]
Closed by #15049
